Reading List/Reading List/Views/BookDetailViewController.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  BookDetailViewController.swift
+//  Reading List
+//
+//  Created by Angel Buenrostro on 1/18/19.
+//  Copyright © 2019 Lambda School. All rights reserved.
+//
+
+import UIKit
+
+class BookDetailViewController: UIViewController {
+    
+    var bookController: BookController?
+    var book: Book?
+
+    @IBOutlet weak var textField: UITextField!
+    @IBOutlet weak var textView: UITextView!
+    @IBAction func saveButtonTapped(_ sender: Any) {
+        guard let book = book else {
+            guard let title = textField.text, let reason = textView.text  else { return }
+            bookController?.Create(title: title, reason: reason)
+            bookController?.saveToPersistentStore()
+            navigationController?.popViewController(animated: true)
+            return
+        }
+        guard let title = textField.text, let reason = textView.text  else { return }
+        bookController?.updateTitleOrReason(for: book, newTitle: title , newReason: reason)
+        bookController?.saveToPersistentStore()
```

Comment:
You already save `updateTitleOrReason` function's result to persistent store inside of `BookController` class implementation
 ```suggestion

```